src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/AttributeRegistryApiServiceImpl.scala

```diff
@@ -0,0 +1,159 @@
+package it.pagopa.interop.attributeregistryprocess.api.impl
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.server.Directives.onComplete
+import akka.http.scaladsl.server.Route
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.attributeregistryprocess.api.AttributeApiService
+import it.pagopa.interop.attributeregistryprocess.api.types.AttributeRegistryServiceTypes._
+import it.pagopa.interop.attributeregistryprocess.error.ResponseHandlers._
+import it.pagopa.interop.attributeregistryprocess.model.{Attribute, AttributeSeed, AttributesResponse, Problem}
+import it.pagopa.interop.attributeregistryprocess.service._
+import it.pagopa.interop.commons.jwt.{ADMIN_ROLE, API_ROLE, authorize}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.commons.utils.TypeConversions._
+import it.pagopa.interop.commons.utils.service.{OffsetDateTimeSupplier, UUIDSupplier}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+final case class AttributeRegistryApiServiceImpl(
+  attributeRegistryManagementService: AttributeRegistryManagementService,
+  uuidSupplier: UUIDSupplier,
+  dateTimeSupplier: OffsetDateTimeSupplier
+)(implicit ec: ExecutionContext)
+    extends AttributeApiService {
+
+  private implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  override def createAttribute(attributeSeed: AttributeSeed)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerAttribute: ToEntityMarshaller[Attribute],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE, API_ROLE) {
+    val operationLabel: String = s"Creating attribute with name ${attributeSeed.name}"
+    logger.info(operationLabel)
+
+    val result: Future[Attribute] = for {
+      createdAttribute <- attributeRegistryManagementService.createAttribute(attributeSeed.toClient).map(_.toApi)
+    } yield createdAttribute
+
+    onComplete(result) {
+      createAttributeResponse[Attribute](operationLabel) { res =>
+        logger.info(s"Attribute created with id ${res.id}")
+        createAttribute201(res)
+      }
+    }
+  }
+
+  override def deleteAttributeById(
+    attributeId: String
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route =
+    authorize(ADMIN_ROLE, API_ROLE) {
+      val operationLabel: String = s"Deleting attribute with ID $attributeId"
+      logger.info(operationLabel)
+
+      val result: Future[Unit] = for {
+        attributeUUID <- attributeId.toFutureUUID
+        result        <- attributeRegistryManagementService.deleteAttributeById(attributeUUID)
```

Comment:
@beetlecrunch shouldn't we check if the attribute is actually used before deleting it?